src/pybind/mgr/selftest/module.py

```diff
@@ -367,6 +368,30 @@ def _self_test_perf_counters(self) -> None:
         # get_counter
         # get_all_perf_coutners
 
+    def _self_test_osd_command(self) -> None:
+        # The Telemetry module depends on the osd_command function,
+        # as well as the command dicts used in this test function.
+
+        # Test 1: `ceph osd.0 perf histogram dump`
+        cmd_dict = {
+                'prefix': 'perf histogram dump',
+                'id': '0',
+                'format': 'json'
+        }
+        r, outb, outs = self.osd_command(cmd_dict)
+        assert r == 0
```

Comment:
Something I noticed while I was implementing this test is that the `r` value (which stands for "return code") always returns 0, even when something in the `cmd_dict` isn't correct (i.e. writing `per` instead of `perf`). If there is an incorrect `cmd_dict` passed to the `mon_command()` function, it will return an `r` value of `-22`, signaling that the `cmd_dict` is incorrect.

I looked into why this is, as I implemented `osd_command()` almost identically to `mon_command()`, and I'm wondering if it has anything to do with the implementation in `send_command()`, the implementation of which can be traced back to `BaseMgrModule`: https://github.com/ceph/ceph/blob/e43a888e395a868bdf888361231e6f431501abfd/src/mgr/BaseMgrModule.cc#L119-L255

I'm looking into the issue, but let me know if anyone else has an idea of why `osd_command()` always comes up with a `0` return code, when it should output `-22` if the `cmd_dict` is incorrect.